papaya-core/src/main/java/pl/edu/agh/papaya/repository/SprintRepository.java

```diff
@@ -30,4 +34,11 @@
 
     @Query("select s from Sprint s where s.dateClosed is null or s.dateClosed > ?1")
     List<Sprint> findNotClosed(LocalDateTime localDateTime);
+
+    default Optional<Sprint> findLastInProject(Long projectId) {
+        return OptionalUtil.fromCollection(findInProjectOrderedFromNewest(projectId, PageRequest.of(0, 1)));
+    }
+
+    @Query("select s from Sprint s where s.project.id = ?1 order by s.durationPeriod.start desc")
+    List<Sprint> findInProjectOrderedFromNewest(Long projectId, Pageable limit);
```

Comment:
Nie wyszłoby nam to samo bez `@Query` tym takim tym:
`List<Sprint> findByProjectIdOrderByDurationPeriodStartDesc(Long projectId, Pageable limit);`?

Znaczy nie mówię, że tak jest lepiej, tylko jestem dumna, że udało mi się poskładać nazwę tej metody jak z klocków i IntelliJ przestał podkreślać, więc chyba jest ok

Reply:
Chociaż generalnie `findInProjectOrderedFromNewest` mi się nie podoba - "najnowsze" brzmi jak "najbardziej niedawno stworzone", a chodzi o "najbardziej w przyszłości", a nie umiem wymyślić lepszej, także może magia Spring Boota nam tutaj pomoże

Reply:
`findInProjectOrderedFromLatest`?

Reply:
`findInProjectOrderedFromLatestStartDate`? Też długie, ale ciężko to opisać krócej, żeby coś mówiło